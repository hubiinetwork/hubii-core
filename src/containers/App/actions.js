import { SUPPORTED_NETWORKS } from 'config/constants';
import {
  NOTIFY,
  CHANGE_NETWORK,
} from './constants';


export function notify(messageType, message) {
  return {
    type: NOTIFY,
    messageType,
    message,
  };
}

export function changeNetwork(name) {
  return {
    type: CHANGE_NETWORK,
    network: SUPPORTED_NETWORKS[name],
  };
}
