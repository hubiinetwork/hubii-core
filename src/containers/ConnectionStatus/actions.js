import {
  NETWORK_FAILURE,
  NETWORK_RECONNECTED,
} from './constants';

export function networkFailure(errorType) {
  return {
    type: NETWORK_FAILURE,
    errorType,
  };
}

export function networkReconnected(errorType) {
  return {
    type: NETWORK_RECONNECTED,
    errorType,
  };
}
