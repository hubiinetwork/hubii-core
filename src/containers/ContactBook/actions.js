/*
 *
 * contacts actions
 *
 */

import {
  CREATE_CONTACT,
  REMOVE_CONTACT,
  EDIT_CONTACT,
} from './constants';

export function createContact(name, address) {
  return {
    type: CREATE_CONTACT,
    contactDetails: { name, address },
  };
}

export function removeContact(contacts, contact) {
  const remainingContacts = contacts.filter(
    (person) =>
    !(person.address === contact.address && person.name === contact.name)
  );
  return {
    type: REMOVE_CONTACT,
    remainingContacts,
  };
}

export function editContact(contacts, newContact, oldContact) {
  const newContactsList = contacts.map(
    (contact) =>
    contact.name === oldContact.name && contact.address === oldContact.address ?
    newContact :
    contact
  );
  return {
    type: EDIT_CONTACT,
    newContactsList,
  };
}
