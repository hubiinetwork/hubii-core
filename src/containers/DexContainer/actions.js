import {
  LOAD_PRICE_HISTORY,
  LOAD_PRICE_HISTORY_SUCCESS,
  LOAD_PRICE_HISTORY_ERROR,
  LISTEN_LATEST_PRICE,
  UPDATE_LATEST_PRICE_SUCCESS,
  UPDATE_LATEST_PRICE_ERROR,
} from './constants';


export function loadPriceHistory(fromCcy) {
  return {
    type: LOAD_PRICE_HISTORY,
    currency: fromCcy,
  };
}
export function loadPriceHistorySuccess(fromCcy, data) {
  return {
    type: LOAD_PRICE_HISTORY_SUCCESS,
    currency: fromCcy,
    data,
  };
}
export function loadPriceHistoryError(fromCcy) {
  return {
    type: LOAD_PRICE_HISTORY_ERROR,
    currency: fromCcy,
  };
}
export function listenLatestPrice(fromCcy) {
  return {
    type: LISTEN_LATEST_PRICE,
    currency: fromCcy,
  };
}
export function updateLatestPriceSuccess(fromCcy, price) {
  return {
    type: UPDATE_LATEST_PRICE_SUCCESS,
    currency: fromCcy,
    price,
  };
}
export function updateLatestPriceError(fromCcy, error) {
  return {
    type: UPDATE_LATEST_PRICE_ERROR,
    currency: fromCcy,
    error,
  };
}
