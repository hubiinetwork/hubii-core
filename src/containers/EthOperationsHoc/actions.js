/*
 *
 * EthOperationsHoc actions
 *
 */

import {
  LOAD_BLOCK_HEIGHT,
  LOAD_BLOCK_HEIGHT_SUCCESS,
  LOAD_BLOCK_HEIGHT_ERROR,
  LOAD_GAS_STATISTICS,
  LOAD_GAS_STATISTICS_SUCCESS,
  LOAD_GAS_STATISTICS_ERROR,
} from './constants';

export function loadBlockHeight() {
  return {
    type: LOAD_BLOCK_HEIGHT,
  };
}

export function loadBlockHeightSuccess(blockHeight) {
  return {
    type: LOAD_BLOCK_HEIGHT_SUCCESS,
    blockHeight,
  };
}
export function loadBlockHeightError(error) {
  return {
    type: LOAD_BLOCK_HEIGHT_ERROR,
    error,
  };
}
export function loadGasStatistics() {
  return {
    type: LOAD_GAS_STATISTICS,
  };
}
export function loadGasStatisticsSuccess(gasStatistics) {
  return {
    type: LOAD_GAS_STATISTICS_SUCCESS,
    gasStatistics,
  };
}
export function loadGasStatisticsError(error) {
  return {
    type: LOAD_GAS_STATISTICS_ERROR,
    error,
  };
}
