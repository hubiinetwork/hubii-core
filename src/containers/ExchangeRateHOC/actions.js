import {
  LOAD_EXCHANGE_RATE,
  LOAD_EXCHANGE_RATE_SUCCESS,
  LOAD_EXCHANGE_RATE_FAILURE,
  LOAD_TICKERS_SUCCESS,
  LOAD_TICKERS_FAILURE,
} from './constants';

export function loadExchangeRate({ ticker, convert }) {
  return {
    type: LOAD_EXCHANGE_RATE,
    ticker,
    convert,
  };
}

export function exchangeRateLoaded({ tickerData, convert }) {
  return {
    type: LOAD_EXCHANGE_RATE_SUCCESS,
    tickerData,
    convert,
  };
}

export function exchangeRateLoadingError({ error, ticker, convert }) {
  return {
    type: LOAD_EXCHANGE_RATE_FAILURE,
    error,
    ticker,
    convert,
  };
}

export function tickersLoaded(tickers) {
  return {
    type: LOAD_TICKERS_SUCCESS,
    tickers,
  };
}

export function tickersLoadingError(error) {
  return {
    type: LOAD_TICKERS_FAILURE,
    error,
  };
}
