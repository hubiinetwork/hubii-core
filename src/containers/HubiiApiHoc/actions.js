/*
 *
 * HubiiApiHoc actions
 *
 */

import {
  LOAD_WALLET_BALANCES,
  LOAD_WALLET_BALANCES_SUCCESS,
  LOAD_WALLET_BALANCES_ERROR,
  LOAD_SUPPORTED_TOKENS,
  LOAD_SUPPORTED_TOKENS_SUCCESS,
  LOAD_SUPPORTED_TOKENS_ERROR,
  LOAD_PRICES,
  LOAD_PRICES_SUCCESS,
  LOAD_PRICES_ERROR,
  LOAD_TRANSACTIONS,
  LOAD_TRANSACTIONS_SUCCESS,
  LOAD_TRANSACTIONS_ERROR,
  LOAD_IDENTITY_SERVICE_TOKEN_SUCCESS,
} from './constants';

export function loadWalletBalances(address, noPoll, onlyEth) {
  return {
    type: LOAD_WALLET_BALANCES,
    address,
    noPoll,
    onlyEth,
  };
}

export function loadWalletBalancesSuccess(address, assets) {
  return {
    type: LOAD_WALLET_BALANCES_SUCCESS,
    address,
    assets,
  };
}

export function loadWalletBalancesError(address, error) {
  return {
    type: LOAD_WALLET_BALANCES_ERROR,
    address,
    error,
  };
}

export function loadSupportedTokens() {
  return {
    type: LOAD_SUPPORTED_TOKENS,
  };
}

export function loadSupportedTokensSuccess(tokens) {
  return {
    type: LOAD_SUPPORTED_TOKENS_SUCCESS,
    assets: [...tokens],
  };
}

export function loadSupportedTokensError(error) {
  return {
    type: LOAD_SUPPORTED_TOKENS_ERROR,
    error,
  };
}

export function loadPrices() {
  return {
    type: LOAD_PRICES,
  };
}

export function loadPricesSuccess(prices) {
  return {
    type: LOAD_PRICES_SUCCESS,
    prices,
  };
}

export function loadPricesError(error) {
  return {
    type: LOAD_PRICES_ERROR,
    error,
  };
}

export function loadTransactions(address) {
  return {
    type: LOAD_TRANSACTIONS,
    address,
  };
}

export function loadTransactionsSuccess(address, transactions) {
  return {
    type: LOAD_TRANSACTIONS_SUCCESS,
    transactions,
    address,
  };
}

export function loadTransactionsError(address, error) {
  return {
    type: LOAD_TRANSACTIONS_ERROR,
    address,
    error,
  };
}

export function loadIdentityServiceTokenSuccess(token) {
  return {
    type: LOAD_IDENTITY_SERVICE_TOKEN_SUCCESS,
    token,
  };
}
