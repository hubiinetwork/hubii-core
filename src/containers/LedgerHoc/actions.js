/*
 *
 * LedgerHoc actions
 *
 */

import { CREATE_WALLET_SUCCESS } from 'containers/WalletHoc/constants';
import getFriendlyError from 'utils/friendlyErrors';

import {
  INIT_LEDGER,
  FETCH_LEDGER_ADDRESSES,
  FETCHED_LEDGER_ADDRESS,
  LEDGER_CONNECTED,
  LEDGER_DISCONNECTED,
  LEDGER_ETH_CONNECTED,
  LEDGER_ETH_DISCONNECTED,
  LEDGER_ERROR,
  LEDGER_CONFIRM_TX_ON_DEVICE,
  LEDGER_CONFIRM_TX_ON_DEVICE_DONE,
} from './constants';


export function initLedger() {
  return {
    type: INIT_LEDGER,
  };
}

export function fetchLedgerAddresses(pathTemplate, firstIndex, lastIndex, hardened) {
  return {
    type: FETCH_LEDGER_ADDRESSES,
    pathTemplate,
    firstIndex,
    lastIndex,
    hardened,
  };
}

export function fetchedLedgerAddress(derivationPath, address) {
  return {
    type: FETCHED_LEDGER_ADDRESS,
    derivationPath,
    address,
  };
}

export function saveLedgerAddress(name, derivationPath, deviceId, address) {
  const newWallet = {
    deviceId,
    address,
    type: 'lns',
    name,
    derivationPath,
  };
  return {
    type: CREATE_WALLET_SUCCESS,
    newWallet,
  };
}

export function ledgerConnected(descriptor) {
  return {
    type: LEDGER_CONNECTED,
    descriptor,
  };
}

export function ledgerDisconnected(descriptor) {
  return {
    type: LEDGER_DISCONNECTED,
    descriptor,
  };
}

export function ledgerEthAppConnected(descriptor, id) {
  return {
    type: LEDGER_ETH_CONNECTED,
    descriptor,
    id,
  };
}

export function ledgerEthAppDisconnected() {
  return {
    type: LEDGER_ETH_DISCONNECTED,
  };
}

export function ledgerError(rawError) {
  const friendlyError = getFriendlyError(rawError);
  return {
    type: LEDGER_ERROR,
    error: friendlyError,
  };
}

export function ledgerConfirmTxOnDevice() {
  return {
    type: LEDGER_CONFIRM_TX_ON_DEVICE,
  };
}

export function ledgerConfirmTxOnDeviceDone() {
  return {
    type: LEDGER_CONFIRM_TX_ON_DEVICE_DONE,
  };
}
