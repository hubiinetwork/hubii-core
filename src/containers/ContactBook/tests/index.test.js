import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { ContactBook, mapDispatchToProps } from '../index';
import { removeContact, editContact } from '../actions';


describe('<ContactBook />', () => {
  describe('mapDispatchToProps', () => {
    describe('removeContact', () => {
      it('should call dispatch', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.removeContact();
        expect(dispatch).toHaveBeenCalledWith(removeContact());
      });
    });
    describe('editContact', () => {
      it('should call dispatch', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.editContact();
        expect(dispatch).toHaveBeenCalledWith(editContact());
      });
    });
  });

  const params = {
    contacts: fromJS([
      {
        name: 'mike',
        address: '0x3123123',
      },
      {
        name: 'joe',
        address: '0x123123123',
      },
    ]),
    removeContact: jest.fn(),
    editContact: jest.fn(),
  };

  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper = shallow(
      <ContactBook
        {...params}
      />
    );
    instance = wrapper.instance();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('shallow mount', () => {
    it('should correctly render', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('methods in <ContactBook />', () => {
    describe('onDelete', () => {
      it('should execute onDelete action', () => {
        const data = {
          name: 'mike',
          address: '0x324324234',
        };
        instance.onDelete(data);
        expect(params.removeContact).toBeCalled();
      });
    });

    describe('filterSearchText', () => {
      it('if type is recentFilterText should filter data based on recentFilterText', () => {
        const state = {
          recentFilterText: 'mike',
          fullFilterText: 'john',
        };
        const data = [
          {
            name: 'mike',
            address: '0x324324234',
          },
          {
            name: 'john',
            address: '0x324s14234',
          },
          {
            name: 'snow',
            address: '0x3243d4234',
          },
        ];
        const type = 'recentFilterText';
        const expectedResult = [
          {
            name: 'mike',
            address: '0x324324234',
          },
        ];
        instance.setState({ ...state });
        expect(instance.filterSearchText(data, type)).toEqual(expectedResult);
      });
      it('if type is fullFilterText should filter data based on fullFilterText', () => {
        const state = {
          recentFilterText: 'mike',
          fullFilterText: 'john',
        };
        const data = [
          {
            name: 'mike',
            address: '0x324324234',
          },
          {
            name: 'john',
            address: '0x324s14234',
          },
          {
            name: 'snow',
            address: '0x3243d4234',
          },
        ];
        const type = 'fullFilterText';
        const expectedResult = [
          {
            name: 'john',
            address: '0x324s14234',
          },
        ];
        instance.setState({ ...state });
        expect(instance.filterSearchText(data, type)).toEqual(expectedResult);
      });
      it('if type is not "recentFilterText" OR "fullFilterText", should return data unfiltered', () => {
        const state = {
          recentFilterText: 'mike',
          fullFilterText: 'john',
        };
        const data = [
          {
            name: 'mike',
            address: '0x324324234',
          },
          {
            name: 'john',
            address: '0x324s14234',
          },
          {
            name: 'snow',
            address: '0x3243d4234',
          },
        ];
        const type = '';
        instance.setState({ ...state });
        expect(instance.filterSearchText(data, type)).toEqual(data);
      });
    });
  });
});

