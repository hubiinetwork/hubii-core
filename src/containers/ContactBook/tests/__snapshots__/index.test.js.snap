// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBook /> shallow mount should correctly render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactBook
    contacts={
        Immutable.List [
            Immutable.Map {
                name: "mike",
                address: "0x324234",
            },
            Immutable.Map {
                name: "john",
                address: "04234",
            },
            Immutable.Map {
                name: "ester",
                address: "0x344234",
            },
        ]
    }
    editContact={[Function]}
    recentContacts={
        Immutable.List [
            Immutable.Map {
                name: "mike",
                address: "0x324234",
            },
            Immutable.Map {
                name: "john",
                address: "04234",
            },
        ]
    }
    removeContact={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <index.style__InnerWrapper1
          contactsPresent={3}
>
          <ContactHeader
                    onChange={[Function]}
                    placeholder="Filter"
                    showSearch={true}
                    title="Recently Used Contacts"
          />
          <index.style__Border
                    contactsLength={2}
          >
                    <ContactList
                              data={
                                        Array [
                                                  Object {
                                                    "address": "0x324234",
                                                    "name": "mike",
                                                  },
                                                  Object {
                                                    "address": "04234",
                                                    "name": "john",
                                                  },
                                                ]
                              }
                              layout="horizontal"
                              message="You have no recently used contacts."
                              onDelete={[Function]}
                              onEdit={[Function]}
                              size="small"
                    />
          </index.style__Border>
</index.style__InnerWrapper1>,
        <index.style__InnerWrapper2
          contactsPresent={3}
>
          <ContactHeader
                    onChange={[Function]}
                    placeholder="Filter"
                    showSearch={true}
                    title="All Contacts"
          />
          <index.style__Border
                    contactsLength={3}
          >
                    <ContactList
                              data={
                                        Array [
                                                  Object {
                                                    "address": "0x324234",
                                                    "name": "mike",
                                                  },
                                                  Object {
                                                    "address": "04234",
                                                    "name": "john",
                                                  },
                                                  Object {
                                                    "address": "0x344234",
                                                    "name": "ester",
                                                  },
                                                ]
                              }
                              layout="horizontal"
                              message="You have not added any contacts."
                              onDelete={[Function]}
                              onEdit={[Function]}
                              size="small"
                    />
          </index.style__Border>
</index.style__InnerWrapper2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ContactHeader
              onChange={[Function]}
              placeholder="Filter"
              showSearch={true}
              title="Recently Used Contacts"
/>,
            <index.style__Border
              contactsLength={2}
>
              <ContactList
                            data={
                                          Array [
                                                        Object {
                                                          "address": "0x324234",
                                                          "name": "mike",
                                                        },
                                                        Object {
                                                          "address": "04234",
                                                          "name": "john",
                                                        },
                                                      ]
                            }
                            layout="horizontal"
                            message="You have no recently used contacts."
                            onDelete={[Function]}
                            onEdit={[Function]}
                            size="small"
              />
</index.style__Border>,
          ],
          "contactsPresent": 3,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Filter",
              "showSearch": true,
              "title": "Recently Used Contacts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ContactList
                data={
                                Array [
                                                Object {
                                                  "address": "0x324234",
                                                  "name": "mike",
                                                },
                                                Object {
                                                  "address": "04234",
                                                  "name": "john",
                                                },
                                              ]
                }
                layout="horizontal"
                message="You have no recently used contacts."
                onDelete={[Function]}
                onEdit={[Function]}
                size="small"
/>,
              "contactsLength": 2,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "address": "0x324234",
                    "name": "mike",
                  },
                  Object {
                    "address": "04234",
                    "name": "john",
                  },
                ],
                "layout": "horizontal",
                "message": "You have no recently used contacts.",
                "onDelete": [Function],
                "onEdit": [Function],
                "size": "small",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ContactHeader
              onChange={[Function]}
              placeholder="Filter"
              showSearch={true}
              title="All Contacts"
/>,
            <index.style__Border
              contactsLength={3}
>
              <ContactList
                            data={
                                          Array [
                                                        Object {
                                                          "address": "0x324234",
                                                          "name": "mike",
                                                        },
                                                        Object {
                                                          "address": "04234",
                                                          "name": "john",
                                                        },
                                                        Object {
                                                          "address": "0x344234",
                                                          "name": "ester",
                                                        },
                                                      ]
                            }
                            layout="horizontal"
                            message="You have not added any contacts."
                            onDelete={[Function]}
                            onEdit={[Function]}
                            size="small"
              />
</index.style__Border>,
          ],
          "contactsPresent": 3,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Filter",
              "showSearch": true,
              "title": "All Contacts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ContactList
                data={
                                Array [
                                                Object {
                                                  "address": "0x324234",
                                                  "name": "mike",
                                                },
                                                Object {
                                                  "address": "04234",
                                                  "name": "john",
                                                },
                                                Object {
                                                  "address": "0x344234",
                                                  "name": "ester",
                                                },
                                              ]
                }
                layout="horizontal"
                message="You have not added any contacts."
                onDelete={[Function]}
                onEdit={[Function]}
                size="small"
/>,
              "contactsLength": 3,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "address": "0x324234",
                    "name": "mike",
                  },
                  Object {
                    "address": "04234",
                    "name": "john",
                  },
                  Object {
                    "address": "0x344234",
                    "name": "ester",
                  },
                ],
                "layout": "horizontal",
                "message": "You have not added any contacts.",
                "onDelete": [Function],
                "onEdit": [Function],
                "size": "small",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <index.style__InnerWrapper1
            contactsPresent={3}
>
            <ContactHeader
                        onChange={[Function]}
                        placeholder="Filter"
                        showSearch={true}
                        title="Recently Used Contacts"
            />
            <index.style__Border
                        contactsLength={2}
            >
                        <ContactList
                                    data={
                                                Array [
                                                            Object {
                                                              "address": "0x324234",
                                                              "name": "mike",
                                                            },
                                                            Object {
                                                              "address": "04234",
                                                              "name": "john",
                                                            },
                                                          ]
                                    }
                                    layout="horizontal"
                                    message="You have no recently used contacts."
                                    onDelete={[Function]}
                                    onEdit={[Function]}
                                    size="small"
                        />
            </index.style__Border>
</index.style__InnerWrapper1>,
          <index.style__InnerWrapper2
            contactsPresent={3}
>
            <ContactHeader
                        onChange={[Function]}
                        placeholder="Filter"
                        showSearch={true}
                        title="All Contacts"
            />
            <index.style__Border
                        contactsLength={3}
            >
                        <ContactList
                                    data={
                                                Array [
                                                            Object {
                                                              "address": "0x324234",
                                                              "name": "mike",
                                                            },
                                                            Object {
                                                              "address": "04234",
                                                              "name": "john",
                                                            },
                                                            Object {
                                                              "address": "0x344234",
                                                              "name": "ester",
                                                            },
                                                          ]
                                    }
                                    layout="horizontal"
                                    message="You have not added any contacts."
                                    onDelete={[Function]}
                                    onEdit={[Function]}
                                    size="small"
                        />
            </index.style__Border>
</index.style__InnerWrapper2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ContactHeader
                onChange={[Function]}
                placeholder="Filter"
                showSearch={true}
                title="Recently Used Contacts"
/>,
              <index.style__Border
                contactsLength={2}
>
                <ContactList
                                data={
                                                Array [
                                                                Object {
                                                                  "address": "0x324234",
                                                                  "name": "mike",
                                                                },
                                                                Object {
                                                                  "address": "04234",
                                                                  "name": "john",
                                                                },
                                                              ]
                                }
                                layout="horizontal"
                                message="You have no recently used contacts."
                                onDelete={[Function]}
                                onEdit={[Function]}
                                size="small"
                />
</index.style__Border>,
            ],
            "contactsPresent": 3,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Filter",
                "showSearch": true,
                "title": "Recently Used Contacts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ContactList
                  data={
                                    Array [
                                                      Object {
                                                        "address": "0x324234",
                                                        "name": "mike",
                                                      },
                                                      Object {
                                                        "address": "04234",
                                                        "name": "john",
                                                      },
                                                    ]
                  }
                  layout="horizontal"
                  message="You have no recently used contacts."
                  onDelete={[Function]}
                  onEdit={[Function]}
                  size="small"
/>,
                "contactsLength": 2,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data": Array [
                    Object {
                      "address": "0x324234",
                      "name": "mike",
                    },
                    Object {
                      "address": "04234",
                      "name": "john",
                    },
                  ],
                  "layout": "horizontal",
                  "message": "You have no recently used contacts.",
                  "onDelete": [Function],
                  "onEdit": [Function],
                  "size": "small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ContactHeader
                onChange={[Function]}
                placeholder="Filter"
                showSearch={true}
                title="All Contacts"
/>,
              <index.style__Border
                contactsLength={3}
>
                <ContactList
                                data={
                                                Array [
                                                                Object {
                                                                  "address": "0x324234",
                                                                  "name": "mike",
                                                                },
                                                                Object {
                                                                  "address": "04234",
                                                                  "name": "john",
                                                                },
                                                                Object {
                                                                  "address": "0x344234",
                                                                  "name": "ester",
                                                                },
                                                              ]
                                }
                                layout="horizontal"
                                message="You have not added any contacts."
                                onDelete={[Function]}
                                onEdit={[Function]}
                                size="small"
                />
</index.style__Border>,
            ],
            "contactsPresent": 3,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Filter",
                "showSearch": true,
                "title": "All Contacts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ContactList
                  data={
                                    Array [
                                                      Object {
                                                        "address": "0x324234",
                                                        "name": "mike",
                                                      },
                                                      Object {
                                                        "address": "04234",
                                                        "name": "john",
                                                      },
                                                      Object {
                                                        "address": "0x344234",
                                                        "name": "ester",
                                                      },
                                                    ]
                  }
                  layout="horizontal"
                  message="You have not added any contacts."
                  onDelete={[Function]}
                  onEdit={[Function]}
                  size="small"
/>,
                "contactsLength": 3,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data": Array [
                    Object {
                      "address": "0x324234",
                      "name": "mike",
                    },
                    Object {
                      "address": "04234",
                      "name": "john",
                    },
                    Object {
                      "address": "0x344234",
                      "name": "ester",
                    },
                  ],
                  "layout": "horizontal",
                  "message": "You have not added any contacts.",
                  "onDelete": [Function],
                  "onEdit": [Function],
                  "size": "small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
